L1Trigger/TrackFindingTMTT/interface/KFTrackletTrack.h

```diff
@@ -0,0 +1,286 @@
+    unsigned int getNumLayers() const { return nLayers_; }
+    // Get number of stubs deleted from track candidate by fitter (because they had large residuals)
+    unsigned int getNumKilledStubs() const { return l1track3D_.getNumStubs() - this->getNumStubs(); }
+
+    // Get bit-encoded hit pattern (where layer number assigned by increasing distance from origin, according to layers track expected to cross).
+    unsigned int getHitPattern() const { return hitPattern_; }
+
+    //--- Get the fitted track helix parameters.
+
+    float qOverPt() const { return qOverPt_; }
+    float charge() const { return (qOverPt_ > 0 ? 1 : -1); }
+    float invPt() const { return fabs(qOverPt_); }
+    float pt() const { return 1. / (1.0e-6 + this->invPt()); }  // includes protection against 1/pt = 0.
+    float d0() const { return d0_; }
+    float phi0() const { return phi0_; }
+    float z0() const { return z0_; }
+    float tanLambda() const { return tanLambda_; }
+    float theta() const { return atan2(1., tanLambda_); }  // Use atan2 to ensure 0 < theta < pi.
+    float eta() const { return -log(tan(0.5 * this->theta())); }
+
+    //--- Get the fitted helix parameters with beam-spot constraint.
+    //--- If constraint not applied (e.g. 4 param fit) then these are identical to unconstrained values.
+
+    bool done_bcon() const { return done_bcon_; }  // Was beam-spot constraint aplied?
+    float qOverPt_bcon() const { return qOverPt_bcon_; }
+    float charge_bcon() const { return (qOverPt_bcon_ > 0 ? 1 : -1); }
+    float invPt_bcon() const { return fabs(qOverPt_bcon_); }
+    float pt_bcon() const { return 1. / (1.0e-6 + this->invPt_bcon()); }
+    float phi0_bcon() const { return phi0_bcon_; }
+
+    // Phi and z coordinates at which track crosses "chosenR" values used by r-phi HT and rapidity sectors respectively.
+    // (Optionally with beam-spot constraint applied).
+    float phiAtChosenR(bool beamConstraint) const {
+      if (beamConstraint) {
+        return reco::deltaPhi(phi0_bcon_ -
+                                  asin((settings_->invPtToDphi() * settings_->chosenRofPhi()) * qOverPt_bcon_) -
+                                  d0_bcon_ / (settings_->chosenRofPhi()),
+                              0.);
+      } else {
+        return reco::deltaPhi(phi0_ - asin((settings_->invPtToDphi() * settings_->chosenRofPhi()) * qOverPt_) -
+                                  d0_ / (settings_->chosenRofPhi()),
+                              0.);
+      }
+    }
+    float zAtChosenR() const {
+      return (z0_ + (settings_->chosenRofZ()) * tanLambda_);
+    }  // neglects transverse impact parameter & track curvature.
+
+    // Get the number of helix parameters being fitted (=5 if d0 is fitted or =4 if d0 is not fitted).
+    float nHelixParam() const { return nHelixParam_; }
+
+    // Get the fit degrees of freedom, chi2 & chi2/DOF
+    unsigned int numDOF() const { return 2 * this->getNumStubs() - nHelixParam_; }
```

Comment:
This is just the standard mathematical expression for the number of degrees of freedom in a track fit to a helix, since the stub can move in two independent directions that would change its distance from the helix. I think the maths here is sufficiently well known that people will recognise it.